### DISCUSSION TITLE
x-data error post loading a submitted form in an HTMX scenario

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#4127] [1. Help]
I have an HTMX form which loads a reschedule-post.html rendered by django.
In my reschedule-post.html I have :

```
<script>
const contact_associate = {{ contact_associate|safe }};
console.log(contact_associate);
</script>

<div x-data="{contacts: contact_associate}">
    <template x-for="(item, index) in contacts" :key="item.pk">
        <span x-text="item.fields.contact_email"></span>
    </template>
</div>
```

The console get printed but I get :

> alpine.min.js:1 
> Alpine Expression Error: contact_associate is not defined
> 
> Expression: "{contacts: contact_associate}"

But note that console.log get printed _after_ the alpineJS error is shown.

When the new HTML code is loaded via HTMX, is it that AlpineJS is not re-initializing or something of that nature ?

**Comment by SimoTod** [comment]
const in js is only scoped for the current block, you can't access it outside of the script tag.

**Comment by ekwoka** [comment]
Also, HTMX Injects the html which Alpine responds to before it handles.ebaluating new script tags.

**Reply by anjanesh** [reply]
So how do I get `x-data="{contacts: contact_associate}"` to work which is available post HTMX load ?

**Reply by ekwoka** [reply]
You can put the content in a script tag as JSON and read it in the data initializer.

Or inject it as JSON with escapejs filter directly in the data initializer.

We mostly use the first.

Like the `|json_tag` filter I think is native to Django

Then parse it in the data.
